cmds/webboot/utils.go

```diff
@@ -51,17 +53,35 @@ func download(URL, fPath string, uiEvents <-chan ui.Event) error {
 	}
 	defer resp.Body.Close()
 
-	f, err := os.Create(fPath)
+	tempFile, err := ioutil.TempFile("", "iso-download-")
 	if err != nil {
 		return err
 	}
-	defer f.Close()
+	defer os.Remove(tempFile.Name())
 
 	go listenForCancel(ctx, cancel, uiEvents)
 	counter := NewWriteCounter(resp.ContentLength)
-	defer counter.Close()
 
-	if _, err = io.Copy(f, io.TeeReader(resp.Body, &counter)); err != nil {
+	if _, err = io.Copy(tempFile, io.TeeReader(resp.Body, &counter)); err != nil {
+		counter.Close()
+		return err
+	}
+
+	counter.Close()
+	copyProgress := menu.NewProgress("Download complete. Writing ISO to cache", true)
+	defer copyProgress.Close()
+
+	if _, err = tempFile.Seek(0, 0); err != nil {
+		return err
+	}
+
+	cacheFile, err := os.Create(fPath)
```

Comment:
Should we check to see if the file is there? if it is there, what should we do?

Reply:
Yeah, if the requested file name matches a file we already have, we can probably avoid downloading it.

Maybe we can check if the file already exists at the beginning of the function?

Reply:
what if it is corrupt? If the system was reboot in the middle of copying to the usb, half a file could be there. Should we give them the option of keeping it or not?

Reply:
or could we do an md5sum and if it fails, remove it?

Reply:
I think the md5sum is a good idea.

Should we remove the existing ISO by default, or maybe show a confirmation screen?

Reply:
I would confirm each step ....
let's see how that looks.

Reply:
I think we can do the file exists checks outside of the download function. I'll put up a PR for that in a bit.